src/client/components/filter-menu/time-filter-menu/preset-time-tab.tsx

```diff
@@ -130,9 +131,13 @@ export class PresetTimeTab extends React.Component<PresetTimeTabProps, PresetTim
   }
 
   private renderLatestPresets() {
+    const { dimension } = this.props;
     const { filterDuration, filterPeriod } = this.state;
-    const presets = LATEST_PRESETS.map(({ name, duration }: TimeFilterPreset) => {
-      return { name, identity: duration };
+    const presets: Array<Preset<string>> = dimension.latestPeriodDurations.map(duration => {
+      return {
+        name: normalizeDurationName(duration.toJS()),
```

Comment:
I'm wondering if there's a reason to keep `Duration` objects in `Dimension` if we only convert it back to `string` every time. Same with `TimeShift`s on `Dimension`

Reply:
Have you got any suggestion, where we could put it instead?

Reply:
I was wondering if we could store `Duration` and `TimeShift` as strings only. But I think that keeping rich objects in our model is better. It is easier to debug and monitor if we see concrete domain objects instead of serialised strings. Just like dates - no one like to debug code and look at ISO date strings :)